Failed to compile MediaQuery property

### Bug Report
* Ray.Di 2.14.0でMediaQueryのQueryオブジェクトをプロパティに持つResourceObjectをコンパイルするとエラーが発生します。
* Ray.Di 2.13.3 では発生しません。
```
% vendor/bin/bear.compile "MyVendor\MyProj" prod-app .
FF..........................................................
............................................................
............

Compilation (1/2) took 1.060000 seconds and used 69.942000MB of memory
Success: 130 Failed: 2
Preload compile: /Users/jingu/MyVendor.MyProj/preload.php (overwritten)
Object graph diagram: /Users/jingu/MyVendor.MyProj
UNBOUND: AssertionError: assert(is_bool($isSingleton)) for MyVendor\MyProj\Resource\App\Article- 
UNBOUND: AssertionError: assert(is_bool($isSingleton)) for MyVendor\MyProj\Resource\App\Articles- 
Compilation (2/2) took 0.040000 seconds and used 5.648000MB of memory
autoload.php: /Users/jingu/MyVendor.MyProj/autoload.php (overwritten)

```

```
% php -v
PHP 8.1.2 (cli) (built: Jan 21 2022 04:47:46) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Xdebug v3.1.3, Copyright (c) 2002-2022, by Derick Rethans
    with Zend OPcache v8.1.2, Copyright (c), by Zend Technologies
```

### How to reproduce

* Resource

```php
<?php

declare(strict_types=1);

namespace MyVendor\MyProj\Resource\App;

use BEAR\Resource\ResourceObject;
use MyVendor\MyProj\Query\ArticleInterface;

class Article extends ResourceObject
{
    public function __construct(private readonly ArticleInterface $article)
    {
    }

    public function onGet(string $id): static
    {
        $this->body = $this->article->item($id);

        return $this;
    }
}
```

* DbQuery
```php
<?php

declare(strict_types=1);

namespace MyVendor\MyProj\Query;

use Ray\MediaQuery\Annotation\DbQuery;

interface CmsArticleInterface
{
    /**
     * @return array<int,array<string,string>>
     */
    #[DbQuery(id: 'article_list')]
    public function list(): array;

    /**
     * @return array<string,string>
     */
    #[DbQuery(id:'article_by_id_item')]
    public function item(string $id): ?array;
}
```

* run compile
```
vendor/bin/bear.compile "MyVendor\MyProj" prod-app .
```

--- a/composer.json
+++ b/composer.json
@@ -19,7 +19,8 @@
         "koriym/printo": "^1.0",
         "nikic/php-parser": "^4.13",
         "ray/aop": "^2.10",
-        "ray/compiler": "^1.6"
+        "ray/compiler": "^1.7"
+
     },
     "require-dev": {
         "ext-pdo": "*",

